src/main/kotlin/com/exactpro/th2/common/schema/configuration/ConfigurationManager.kt

```diff
@@ -15,16 +15,28 @@
 
 package com.exactpro.th2.common.schema.configuration
 
+import com.exactpro.th2.common.ConfigurationProvider
+import com.exactpro.th2.common.Module
+import com.exactpro.th2.common.ModuleFactory
 import com.fasterxml.jackson.databind.ObjectMapper
-import mu.KotlinLogging
-import org.apache.commons.text.StringSubstitutor
 import java.io.IOException
 import java.nio.file.Files
 import java.nio.file.Path
+import java.util.ServiceLoader
 import java.util.concurrent.ConcurrentHashMap
+import mu.KotlinLogging
+import org.apache.commons.text.StringSubstitutor
 
 class ConfigurationManager(private val configurationPath: Map<Class<*>, Path>) {
-    private val configurations: MutableMap<Class<*>, Any?> = ConcurrentHashMap()
+    private val configurationCache: MutableMap<Class<*>, Any?> = ConcurrentHashMap()
+    private val moduleCache: MutableMap<Class<*>, Module> = ConcurrentHashMap()
+    private val modulesFactoryMapping: MutableMap<Class<*>, ModuleFactory> = ConcurrentHashMap()
+
+    init {
+        ServiceLoader.load(ModuleFactory::class.java).forEach { moduleFactory ->
+            modulesFactoryMapping[moduleFactory.moduleType] = moduleFactory
+        }
```

Comment:
Why modules are in the configuration manager? Shouldn't they be in the factory?

Reply:
Maybe move this cache into an abstract part of ConfigurationProvider to avoid new entities

Reply:
After discussion with @Xanclry we have decided to rename ConfigurationManager to ModuleManager